sdk/storage/azure-storage-common/src/storage_retry_policy.cpp

```diff
@@ -124,7 +122,9 @@ namespace Azure { namespace Storage { namespace Details {
 
         if (retryDelayMs != 0)
         {
-          std::this_thread::sleep_for(std::chrono::milliseconds(retryDelayMs));
+          auto delay = std::chrono::milliseconds(retryDelayMs);
```

Comment:
OK, make the function internal (not private) and call it in both retry policies.

Reply:
Sure.

Question:  We can make this function to behave in 2 ways:

Imagine this case:  
 - Create context `C` with 100ms time out.
 - Send request, then reach to a point where we are going to sleep for 500 ms.

First result: Since we know the context will throw, we just do it early and before reaching 100ms.  (this is the actual implementation)

Second result: Sleep for exactly the time left for cancelling the request.  `ThrowIfWillCancelledAfter` will call sleep for any time left to only reach the time out and then it throws. This way we wouldn't throw early and we won't also wait for more than the requested time (when the sleep time is greater than that, like this example)

Reply:
Moved to be Internal API